java/src/com/google/idea/blaze/java/run/ExternalFilePositionManager.java

```diff
@@ -56,11 +56,17 @@ protected PsiFile getPsiFileByLocation(Project project, Location location) {
     return refType != null ? qualifiedNameToPsiFile.get(refType.name()) : null;
   }
 
+
   @Override
   public List<ClassPrepareRequest> createPrepareRequests(
       ClassPrepareRequestor requestor, SourcePosition position) throws NoDataException {
+    List<ClassPrepareRequest> prepareRequests = super.createPrepareRequests(requestor, position);
+    // PrepareRequests returns an empty list for Kotlin sources. Returning this blocks the KotlinManager from contributing.
```

Comment:
> Returning this blocks the KotlinManager from contributing.

The doc says:
>   * @return the prepare request, or null if the code is managed by this {@code PositionManager} but no class prepare notification is needed
>   * @throws NoDataException if the position is not in the code managed by this {@code PositionManager}

How does the built-in position manager avoid this conflict? Is it based on extension point ordering?

Reply:
Can't say much on this one without fully getting into the underlying code. 

I spend a lot of time to track down this problem, and I have been debugging Kotlin and Java code in our mono repo since without any issues - I even break pointed on Scala jars. 

We could be more explicit in this method (and / or the other ones) by testing the language type: --i.e., `position.getElementAt().getLanguage().is(KotlinLanguage.INSTANCE)`. Java would have a link dependency on the Kotlin but not a problem now since it's included in the platform.